packages/transition-frontend/src/app-transition.tsx

```diff
@@ -50,28 +50,41 @@ const contributions = [
     new DemandManagementDashboardContribution(),
     new SupplyDemandAnalysisDashboardContribution()
 ];
-const jsx = (
+
+let hasConfig: boolean | undefined = undefined;
+let hasFetchedAuth: boolean | undefined = undefined;
+let hasRendered = false;
+const jsx = () => (
     <Provider store={store}>
-        <I18nextProvider i18n={i18n}>
+        <I18nextProvider i18n={initI18n()}>
             <Router history={history}>
-                <TransitionRouter contributions={contributions} config={config} mainMap={MainMap as any} />
+                {hasConfig === true ? (
+                    <TransitionRouter contributions={contributions} config={config} mainMap={MainMap as any} />
+                ) : (
+                    <MaintenancePage />
+                )}
             </Router>
         </I18nextProvider>
     </Provider>
 );
 
 ReactDOM.render(<LoadingPage />, document.getElementById('app'));
 
-let hasRendered = false;
 const renderApp = () => {
-    if (!hasRendered) {
-        ReactDOM.render(jsx, document.getElementById('app'));
+    if (!hasRendered && hasConfig !== undefined && hasFetchedAuth !== undefined) {
```

Comment:
In retrospect, 2 things I don't like here: the multiple flags and the fact that we need to call a i18n initialization function. It makes the currentLanguage potentially undefined and multiplies the possibilities. I think we should fetch the configuration, then import the rest which we are certain will run with a valid configuration. If we can't get the config, we put a [eventually] pretty and generic failure page.